realm/src/androidTest/java/io/realm/RealmCursorTest.java

```diff
@@ -0,0 +1,595 @@
+        }
+    }
+
+    public void testGetColumnName() {
+        assertEquals("columnLong", cursor.getColumnName(1));
+    }
+
+    public void testGetColumnNames() {
+        String[] names = cursor.getColumnNames();
+        assertEquals(AllTypes.COL_COUNT, names.length);
+        assertEquals("columnString", names[AllTypes.COL_INDEX_STRING]);
+        assertEquals("columnRealmList", names[AllTypes.COL_INDEX_LIST]);
+    }
+
+    public void testGetColumnCount() {
+        assertEquals(9, cursor.getColumnCount());
+    }
+
+    // Test that all get<type> method throw IndexOutOfBounds properly
+    public void testGetXXXInvalidIndexThrows() {
+        cursor.moveToFirst();
+        for (CursorGetter cursorGetter : CursorGetter.values()) {
+            try {
+                switch (cursorGetter) {
+                    case STRING: cursor.getString(-1); break;
+                    case SHORT: cursor.getShort(-1); break;
+                    case INT: cursor.getInt(-1); break;
+                    case LONG: cursor.getLong(-1); break;
+                    case FLOAT: cursor.getFloat(-1); break;
+                    case DOUBLE: cursor.getDouble(-1); break;
+                    case BLOB: cursor.getBlob(-1); break;
+                }
+                fail(cursorGetter + " should throw an exception");
+            } catch (IndexOutOfBoundsException expected) {
+            } catch (Exception wrongException) {
+                throw new RuntimeException(cursorGetter + " threw the wrong exception: ", wrongException);
+            }
+        }
+    }
+
+    // Test that all get<type> method throw if field type doesn't match getter type
+    public void testGetXXXWrongFieldTypeThrows() {
+        cursor.moveToFirst();
+        for (CursorGetter cursorGetter : CursorGetter.values()) {
+            try {
+                switch (cursorGetter) {
+                    case STRING: cursor.getString(AllTypes.COL_INDEX_LONG); break;
+                    case SHORT: cursor.getShort(AllTypes.COL_INDEX_STRING); break;
+                    case INT: cursor.getInt(AllTypes.COL_INDEX_STRING); break;
+                    case LONG: cursor.getLong(AllTypes.COL_INDEX_STRING); break;
+                    case FLOAT: cursor.getFloat(AllTypes.COL_INDEX_STRING); break;
+                    case DOUBLE: cursor.getDouble(AllTypes.COL_INDEX_STRING); break;
+                    case BLOB: cursor.getBlob(AllTypes.COL_INDEX_STRING); break;
+                }
+                fail(cursorGetter + " should throw an exception");
+            } catch (IllegalArgumentException expected) {
+            } catch (Exception wrongException) {
+                throw new RuntimeException(cursorGetter + " threw the wrong exception: ", wrongException);
+            }
+        }
+    }
+
+    // Test that all getters fail when the cursor is closed
+    public void testGetXXXFailWhenCursorClosed() {
+        cursor.close();
+        for (CursorGetter cursorGetter : CursorGetter.values()) {
```

Comment:
You don't have any checks if the cursor is closed in any methods, so it would expose random NullPointerExceptions everywhere. If that's the intended behaviour there is no reason to actually test it. And here it's also just a NullPointerException anyway. What's the norm with cursors?

It's not that big a thing to test actually, if you use my "compact" form of writing the tests.